### ISSUE TITLE
Add option to change language to default icons

### ISSUE THREAD

**Issue by luna404** [HeRAMS-WHO/herams-backend#592]
If a user wants to change the language, they have to go through profile, etc. This is not intuitive and users are often not aware that they can change the language.

Can we add an icon to the standard icons on the top right corner that provides users with a short cut to change the language. Ideally, we could use country flags to indicate the selected language.

**Comment by luna404** [comment]
This has not been implemented yet but it does work on staging.

**Comment by jeremiehelme** [comment]
Ah, then I misunderstood,

I've added a shortcut to the the page where you can change the language, but you would like a language selector instead.
I agree that would be event better, I'll add it

**Comment by luna404** [comment]
No need to change anything. It's fine the was it is on staging, but I thought the language button wasn't showing on Prod. I'll double check once the platform is back up and comment here.

**Comment by luna404** [comment]
@jeremiehelme if you check Prod, the icon displaying the select language is not visible.

![image](https://user-images.githubusercontent.com/18245059/94655790-69361480-02ff-11eb-9fc6-53dec5f477a5.png)


It's fine on staging 
![image](https://user-images.githubusercontent.com/18245059/94655849-81a62f00-02ff-11eb-8e4b-64d0a0a83695.png)

**Comment by jeremiehelme** [comment]
That's strange then, I can see it on prod too
I notice there is an extra space after the question mark on your screenshot, maybe the language code is empty

Can you right click on it and choose "inspect" ? 
You should see something like this

![Capture d’écran 2020-09-30 à 09 40 06](https://user-images.githubusercontent.com/2345926/94656892-fa59bb00-0300-11eb-8786-9f1e07ba2e71.png)

Is it empty ?

**Comment by luna404** [comment]
My bad, you're absolutely right, I had no language selected. 

I assume most users will be in the same situation. Are you able to get the default language a uses is using?

**Comment by jeremiehelme** [comment]
The language variable has a default value, it shouldn't be empty, I'll ask Sam about it

**Comment by jeremiehelme** [comment]
@SamMousa to explain the bug  

I used `\Yii::$app->language;` to display the user language in the user menu (view/user-menu)
On production, Caroline had an empty value. She had to update her account for the language to be set correctly
Now it displays well

`\Yii::$app->language;` is set by default to `'en-US'` by Yii, 
In config/common, after the session is created, it is set to `$event->identity->language;` if not empty

I wonder why it was not set correctly for Caroline 
@luna404 Do you know if other users don't see the language in the user menu too ?


Thanks for your help,

**Comment by luna404** [comment]
@jeremiehelme yes, it's same issue for any users that changes the language to auto-detect.

**Comment by jeremiehelme** [comment]
Ok thanks ! 

So \Yii::$app->language is empty when autodetect is chosen
I'll check why

**Comment by SamMousa** [comment]
It shouldn't be, it goes to English for me..

**Comment by jeremiehelme** [comment]
On the user account page, in the language selector, the <option> element for the auto-detected language has no value
Is there a reason for it ?

Selecting it causes `\Yii::$app->language` to be empty

For a newly created account, the language is set correctly, It's just when people choose that option that the language is unsetted

**Comment by SamMousa** [comment]
That's how the element works. Autodetect will work when no value is set. If a language is set then that language is used instead of auto detection

**Comment by luna404** [comment]
@jeremiehelme have you made any progress on this?

**Comment by SamMousa** [comment]
I think this also needs to go into our design language discussion...

This is another example of "let's add a random shortcut to simplify one specific action"....
Setting the language is something people do only once, it doesn't make sense to me to make this a primary navigational entry...

**Comment by luna404** [comment]
No longer relevant. User should get used to use the new "user" icon.
